python/styleguide.md

```diff
@@ -0,0 +1,108 @@
+# Python Code Style Guide
+
+
+Documentation
+-------------
+
+* We use doc strings for package, module, class and function documentation.
+    * A doc string is a string that is the first statement in a package, module, class or function.
+    * These strings can be extracted automatically through the __doc__ member of the object. 
+
+* We always use the three double-quote `"""` format for doc strings.
+    * This is defined in [PEP 257](https://www.python.org/dev/peps/pep-0257/)
+
+* A doc string should be organized as
+    * a summary line (one physical line),
+    * a blank line,
+    * the rest of the doc string.
```

Comment:
Idea: We should give some examples of what *good* docstrings look like. I personally like what PEP 257 says about that:

> * The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

> * The one-line docstring should NOT be a "signature" reiterating the function/method parameters (which can be obtained by introspection). [...]